src/pages/Admin/WorldUsers/WorldUsers.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useMemo } from "react";
+import { useForm } from "react-hook-form";
+import { Button } from "components/admin/Button";
+import { Header } from "components/admin/Header";
+import { Input } from "components/admin/Input";
+import { InviteAdminModal } from "components/admin/InviteAdminModal";
+import { WorldUserCard } from "components/admin/WorldUserCard";
+import { AdminLayout } from "components/layouts/AdminLayout";
+import { includes } from "lodash";
+
+import { UserId, WorldId } from "types/id";
+
+import { useWorldAndSpaceByParams } from "hooks/spaces/useWorldAndSpaceByParams";
+import { useWorldSpaces } from "hooks/spaces/useWorldSpaces";
+import { useProfileByIds } from "hooks/user/useProfileByIds";
+import { useShowHide } from "hooks/useShowHide";
+import { useUser } from "hooks/useUser";
+
+import { LoadingPage } from "components/molecules/LoadingPage";
+
+import { AdminRestricted } from "components/atoms/AdminRestricted";
+
+import * as TW from "./WorldUsers.tailwind";
+
+export interface WorldScheduleFormInput {
+  startDate: string;
+  startTime: string;
+  endDate: string;
+  endTime: string;
+}
+
+export const WorldUsers: React.FC = () => {
+  const {
+    world,
+    worldId,
+    isLoading: isWorldLoading,
+  } = useWorldAndSpaceByParams();
+
+  const { userId } = useUser();
+
+  const {
+    isShown: isShownInviteAdminModal,
+    show: showInviteAdminModal,
+    hide: hideInviteAdminModal,
+  } = useShowHide();
+
+  const defaultValues = useMemo(
+    () => ({
+      nameSearchQuery: "",
+      spaceSearchQuery: "",
+    }),
+    []
+  );
+
+  const { register, watch } = useForm({
+    reValidateMode: "onChange",
+    defaultValues,
+  });
+
+  const { nameSearchQuery, spaceSearchQuery } = watch();
+
+  const owners = world?.owners;
+
+  const { spaces } = useWorldSpaces({ worldId: worldId ?? ("" as WorldId) });
+
+  const users = useProfileByIds({ userIds: owners as UserId[] });
+
+  const filteredUsers = users?.filter((user) =>
+    user.partyName?.toLowerCase().includes(nameSearchQuery.toLowerCase())
+  );
+
+  const userSpaces = filteredUsers?.map((user) => {
+    return {
+      user: user,
+      spaces: spaces.filter((space) =>
+        space.owners?.includes(user.id as string)
+      ),
+    };
+  });
+
+  const filteredUserSpaces = userSpaces?.filter((userSpace) => {
+    const spacesNames = userSpace.spaces
+      .map((space) => space.name.toLowerCase())
+      .join();
+
+    return includes(spacesNames, spaceSearchQuery.toLowerCase());
+  });
+
+  if (isWorldLoading) {
+    return <LoadingPage />;
+  }
```

Comment:
[OOS] 

There is a solution to not having these checks in a component, especially if we're replacing those with `compose()` by adding a simpler pedestrian HOC. Example:

`WorldUsersHoc.tsx` would have the hook and this kind of code:
```tsx

  if (isWorldLoading) {
    return <LoadingPage />;
  }
  
  return <WorldUsers world={world} />
```

`index.ts` would have 
```ts
export { WorldUsersHoc as WorldUsers } from './WorldUsers';
```